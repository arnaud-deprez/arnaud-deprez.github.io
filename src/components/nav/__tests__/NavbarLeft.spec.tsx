import React from 'react'
import { render } from '@testing-library/react'
import { useStaticQuery } from 'gatsby'
import { Nav as BootstrapNav } from 'react-bootstrap'
import { NavbarLeft, Nav } from '..'

jest.mock('gatsby')

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const mockUseStaticQuery = useStaticQuery as jest.Mock<any, any>

describe('<NavbarLeft/>', () => {
  beforeAll(() => {
    mockUseStaticQuery.mockReturnValue({
      file: {
        childImageSharp: {
          resize: {
            src: 'profile.png',
          },
        },
      },
    })
  })
  it('renders correctly', () => {
    const { asFragment } = render(
      <NavbarLeft
        author={{
          name: 'Arnaud Deprez',
          jobTitle: 'Technical Architect',
          email: undefined,
          linkedin: undefined,
          twitter: undefined,
          github: undefined,
          rss: undefined,
        }}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it('renders correctly with children', () => {
    const { asFragment } = render(
      <NavbarLeft
        author={{
          name: 'Arnaud Deprez',
          jobTitle: 'Technical Architect',
          email: undefined,
          linkedin: undefined,
          twitter: undefined,
          github: undefined,
          rss: undefined,
        }}
      >
        <Nav>
          <BootstrapNav.Link>Menu 1</BootstrapNav.Link>
          <BootstrapNav.Link>Menu 2</BootstrapNav.Link>
          <BootstrapNav.Link>Menu 3</BootstrapNav.Link>
        </Nav>
      </NavbarLeft>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
