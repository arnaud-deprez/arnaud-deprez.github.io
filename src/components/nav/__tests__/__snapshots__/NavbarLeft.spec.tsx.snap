// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavbarLeft/> renders correctly 1`] = `
<DocumentFragment>
  <nav
    class="d-none d-lg-flex flex-column navbar-dark bg-primary navbar navbar-expand navbar-light"
    id="navbar-left"
    role="navigation"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      <div
        class="photo-card card"
      >
        <img
          alt="profile"
          class="card-img rounded-circle mb-2"
          src="profile.png"
        />
        <div
          class="d-none d-lg-block card-body"
        >
          <h5
            class="text-center card-title"
          >
            Arnaud Deprez
          </h5>
          <h6
            class="font-italic text-center text-wrap card-subtitle"
          >
            Technical Architect
          </h6>
        </div>
      </div>
    </a>
    <ul
      class="social-icons mb-3 navbar-nav"
    />
  </nav>
</DocumentFragment>
`;

exports[`<NavbarLeft/> renders correctly with children 1`] = `
<DocumentFragment>
  <nav
    class="d-none d-lg-flex flex-column navbar-dark bg-primary navbar navbar-expand navbar-light"
    id="navbar-left"
    role="navigation"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      <div
        class="photo-card card"
      >
        <img
          alt="profile"
          class="card-img rounded-circle mb-2"
          src="profile.png"
        />
        <div
          class="d-none d-lg-block card-body"
        >
          <h5
            class="text-center card-title"
          >
            Arnaud Deprez
          </h5>
          <h6
            class="font-italic text-center text-wrap card-subtitle"
          >
            Technical Architect
          </h6>
        </div>
      </div>
    </a>
    <ul
      class="social-icons mb-3 navbar-nav"
    />
    <ul
      class="navbar-nav"
    >
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="#"
          role="tab"
          tabindex="-1"
        >
          Menu 1
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="#"
          role="tab"
          tabindex="-1"
        >
          Menu 2
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="#"
          role="tab"
          tabindex="-1"
        >
          Menu 3
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
